### PR TITLE
lint: 🇺🇸🇺🇸🇺🇸 use US locale in TestMisspell 🇺🇸🇺🇸🇺🇸

### PR DESCRIPTION
**Pull request by nvb**
This change adds a US locality preference to `TestMisspell`. This causes
`misspell` to add rules from its `DictAmerican` dictionary which catch
british vs. american spelling differences like "cancelled" vs. "canceled"
and "marshalled" vs. "marshaled".

(I don't feel strongly about "marshalled" vs. "marshaled". We can ignore
that rule if we want and cut down on the size of this change)

The bulk of this change was automated by running the command:
```
git ls-files pkg/ | xargs ./bin/misspell -locale US -i "licence,analyse" -w
```

After this change:
```
Nathans-MBP:cockroach$ git grep -i cancelled | wc -l
       0
```

cc. @petermattis

### CODE REVIEW

**Comment by nvanbenschoten** [comment]
> I don't feel strongly about "marshalled" vs. "marshaled". We can ignore
that rule if we want and cut down on the size of this change.

I'd actually be completely fine with using the UK locale instead (and by extension, using "cancelled" over "canceled"). As long as we're reducing ambiguity, I'm happy.

**Comment by RaduBerinde** [comment]
> I'd actually be completely fine with using the UK locale instead (and by extension, using "cancelled" over "canceled"). As long as we're reducing ambiguity, I'm happy.

Most libraries we depend on use US spelling (e.g. `context.Canceled()`, protobuf `IsMarshaler`), I vote for using the US spelling everywhere.

**Comment by mberhault** [comment]
Definitely not to the UK locale. A simple misspell UK run shows 2K (almost the binary version) changes the vast majority being 's' instead of 'z' (zed). eg: "initialised" instead of "initialized".

My vote is to do nothing. The vast majority of the changes are in comments not impacting public names others may have to worry about.
Having my build fail in TC just because I can't wrap my head around "canceled" is a bit annoying (about as much as the damn line wrapping for functions which has to have the wrap in exactly the right place).

Too much linting of this kind just wastes time for no good reason.

**Comment by RaduBerinde** [comment]
I have no preference on whether we should add a linter for this, but even if we don't I think this is still a good change to merge.

**Comment by petermattis** [comment]
:lgtm:

---

Review status: 0 of 81 files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/20342#-:-L-CQUcle5rg4Qxo1oCh:bnfp4nl)*

**Comment by nvanbenschoten** [comment]
TFTRs. I'm going to merge this as is because the longer it waits, the more it's going to conflict with other changes also in flux. If the extra linting becomes an issue we can re-evaluate, but I really doubt it will become one. We already have `TestMisspell` in our linter suite, this is just adding a few more rules to it.

File: pkg/ui/src/util/api.ts [hunk 1]
```diff
@@ -110,7 +110,7 @@ export function toArrayBuffer(encodedRequest: Uint8Array): ArrayBuffer {
 }
 
 // timeoutFetch is a wrapper around fetch that provides timeout and protocol
-// buffer marshalling and unmarshalling.
+// buffer marshaling and unmarshalling.
```
Comment (couchand) [inline pkg/ui/src/util/api.ts]: this line looks suspicious to me @nvanbenschoten, do we need to explicitly add and error for "unmarshalling"?

Comment (nvanbenschoten) [reply pkg/ui/src/util/api.ts]: Good catch! I opened https://github.com/client9/misspell/issues/126 to track this.

**Pull request merged**